---
title: "PCA_Trial_R"
output: html_document
date: "2024-01-12"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Read the csv file in 
```{r}
df <- read_csv("/Users/shrinivas/Desktop/Paulinella/P_ovalis_Metagenome/Tree_plot/Acr_PCA.csv")
sample_data <- read_csv("/Users/shrinivas/Desktop/Paulinella/P_ovalis_Metagenome/Tree_plot/sample_info.csv")
```

```{r}
log10_safe <- function(x) {
  if(all(is.numeric(x) & x > 0)) {
    return(log10(x))
  } else {
    return(x)
  }
}

# Apply log10 transformation to all columns
df_log10 <- data.frame(lapply(df, log10_safe))
```

```{r}
pca_matrix <- df_log10 %>% 
  # make the "gene" column become the rownames of the table
  column_to_rownames("Protein.IDs") %>% 
  # coerce to a matrix
  as.matrix() %>% 
  # transpose the matrix so that rows = samples and columns = variables
  t()

# Perform the PCA
sample_pca <- prcomp(pca_matrix)
```

```{r}
pca_matrix[1:10, 1:5]
as_tibble(pca_matrix)
as_tibble(pca_matrix, rownames = "sample")
```

```{r}
pc_eigenvalues <- sample_pca$sdev^2
pc_eigenvalues <- tibble(PC = factor(1:length(pc_eigenvalues)), 
                         variance = pc_eigenvalues) %>% 
  # add a new column with the percent variance
  mutate(pct = variance/sum(variance)*100) %>% 
  # add another column with the cumulative variance explained
  mutate(pct_cum = cumsum(pct))

# print the result
pc_eigenvalues
```

```{r}
pc_eigenvalues %>% 
  ggplot(aes(x = PC)) +
  geom_col(aes(y = pct)) +
  geom_line(aes(y = pct_cum, group = 1)) + 
  geom_point(aes(y = pct_cum)) +
  labs(x = "Principal component", y = "Fraction variance explained")

```

```{r}
pc_scores <- sample_pca$x

pc_scores <- pc_scores %>% 
  # convert to a tibble retaining the sample names as a new column
  as_tibble(rownames = "sample")

# print the result
pc_scores
```


```{r}
pc_scores %>% 
  # create the plot
  ggplot(aes(x = PC1, y = PC2)) +
  geom_point()

```


```{r}

# get the PC scores from prcomp object
# get the PC scores from prcomp object
sample_pca$x %>% 
  # convert it to a tibble
  as_tibble(rownames = "Protein.IDs") %>% 
  # join with "sample_info" table
  left_join(sample_data, by = "Protein.IDs") %>%  # Replace "sample_id" with the actual column name to join on
  # make the plot
  ggplot(aes(x = PC1, y = PC2, color = factor(Treatment), shape = factor(Timepoint))) +
  geom_point()

```